systemd/systemd.go

```diff
@@ -1392,22 +1400,37 @@ const (
 	snappyOriginModule = "X-SnapdOrigin"
 )
 
-func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+func ensureMountUnitFile(u *MountUnitOptions) (mountUnitName string, modified mountUpdateStatus, err error) {
 	if u == nil {
-		return "", errors.New("writeMountUnitFile() expects valid mount options")
+		return "", mountUnchanged, errors.New("ensureMountUnitFile() expects valid mount options")
 	}
 
 	mu := MountUnitPathWithLifetime(u.Lifetime, u.Where)
-	outf, err := osutil.NewAtomicFile(mu, 0644, 0, osutil.NoChown, osutil.NoChown)
-	if err != nil {
-		return "", fmt.Errorf("cannot open mount unit file: %v", err)
+	var unitContent bytes.Buffer
+
+	if err := parsedMountUnitTemplate.Execute(&unitContent, &u); err != nil {
+		return "", mountUnchanged, fmt.Errorf("cannot generate mount unit: %v", err)
 	}
-	defer outf.Cancel()
 
-	if err := parsedMountUnitTemplate.Execute(outf, &u); err != nil {
-		return "", fmt.Errorf("cannot generate mount unit: %v", err)
+	if osutil.FileExists(mu) {
+		modified = mountUpdated
+	} else {
+		modified = mountCreated
+	}
+
+	os.MkdirAll(filepath.Dir(mu), 0755)
```

Comment:
Is here an error check `if err := os.MkidrAll(...` missing?

Reply:
It is because it does not matter much. It is quite common to not care about failure from a recursive mkdir. But looking the rest of snapd, this project always checks. I can add the check.